graphql-ballerina/Module.md

```diff
@@ -0,0 +1,685 @@
+
+```ballerina
+public type Person record {|
+    string name;
+    int age;
+|};
+
+resource function get profile(int id) returns Person? {
+    if (id == 1) {
+        return { name: "Walter White", age: 51 };
+    }
+}
+```
+
+This will generate the following schema:
+
+```
+type Query {
+    profile: Person
+}
+
+type Person {
+    name: String!
+    age: Int!
+}
+```
+
+This can be queried using the following document:
+```
+{
+    profile(id: 1) {
+        name
+    }
+}
+```
+
+Result:
+```json
+{
+    "data": {
+        "profile": {
+            "name": "Walter White"
+        }
+    }
+}
+```  
+
+If the following document is used:
+```
+{
+    profile(id: 4) {
+        name
+    }
+}
+```
+
+This will be the result:
+```json
+{
+    "data": {
+        "profile": null
+    }
+}
+``` 
+
+#### Union Types
+The Ballerina GraphQL service can return a union of distinct service types. This will be mapped to a GraphQL `UNION` type.
+
+Since Ballerina supports the union types by nature, directly returning a union type is also allowed, but not recommended. The recommended way is to define a union type name separately and then use that type name, as shown in the following example. If a union type is returned directly without providing a type name, the type name will be `T1|T2|T3|...|Tn`.
+
+```ballerina
+import ballerina/graphql;
+
+public type StudentOrTeacher Student|Teacher;
+
+service /graphql on new graphql:Listener(4000) {
+    resource function get profile(int purity) returns StudentOrTeacher {
+        if (purity < 90) {
+            return new Student(1, "Jesse Pinkman");
+        } else {
+            return new Teacher(737, "Walter White", "Chemistry");
+        }
+    }
+}
+
+distinct service class Student {
+    private int id;
+    private string name;
+    
+    public function init(int id, string name) {
+        self.id = id;
+        self.name = name;
+    }
+    
+    resource function get id() returns int {
+        return self.id;
+    }
+    
+    resource function get name() returns string {
+        return self.name;
+    }
+}
+
+distinct service class Teacher {
+    private int id;
+    private string name;
+    private string subject;
+    
+    public function init(int id, string name, string subject) {
+        self.id = id;
+        self.name = name;
+        self.subject = subject;
+    }
+    
+    resource function get id() returns int {
+        return self.id;
+    }
+    
+    resource function get name() returns string {
+        return self.name;
+    }
+        
+    resource function get subject() returns string {
+        return self.subject;
+    }
+}
+```
+
+This will generate the following schema: 
+
+```
+type Query {
+    profile(purity: Int!): StudentOrTeacher!
+}
+
+type Student {
+    id: Int!
+    name: String!
+}
+
+type Teacher {
+    id: Int!
+    name: String!
+    subject: String!
+}
+
+union StudentOrTeacher Student|Teacher
+```
+
+This can be queried using the following document:
```

Comment:
```suggestion
This can be queried using the following document:
```
Is it correct to say "the following document"? Or, should it be "the following code"? Please check all occurrences.

Reply:
Same as above